packages/ai/ai/src/AiInput.ts

```diff
@@ -1,227 +1,219 @@
 /**
  * @since 1.0.0
  */
-import type { PlatformError } from "@effect/platform/Error"
-import * as FileSystem from "@effect/platform/FileSystem"
-import * as Path from "@effect/platform/Path"
-import * as Chunk from "effect/Chunk"
-import * as Context from "effect/Context"
-import * as Effect from "effect/Effect"
-import * as Encoding from "effect/Encoding"
-import { dual } from "effect/Function"
-import * as Option from "effect/Option"
-import * as ParseResult from "effect/ParseResult"
 import * as Predicate from "effect/Predicate"
-import * as Schema_ from "effect/Schema"
-import { AiResponse, ToolCallId, WithResolved } from "./AiResponse.js"
-import * as AiRole from "./AiRole.js"
+import * as Schema from "effect/Schema"
+import * as AiResponse from "./AiResponse.js"
+import * as InternalCommon from "./internal/common.js"
 
 const constDisableValidation = { disableValidation: true } as const
 
 /**
  * @since 1.0.0
- * @category parts
+ * @category Models
  */
-export const PartTypeId: unique symbol = Symbol("@effect/ai/AiInput/Part")
+export const AiInput: Schema.Array$<Schema.suspend<Message, typeof Message.Encoded, never>> = Schema.Array(
+  Schema.suspend(() => Message)
+).pipe(Schema.annotations({ identifier: "@effect/ai/AiInput" }))
 
 /**
  * @since 1.0.0
- * @category parts
+ * @category Models
  */
-export type PartTypeId = typeof PartTypeId
+export const AiInputFromJson: Schema.Schema<ReadonlyArray<Message>, string> = Schema.parseJson(AiInput)
 
 /**
  * @since 1.0.0
- * @category parts
+ * @category Models
  */
-export class TextPart extends Schema_.TaggedClass<TextPart>("@effect/ai/AiInput/TextPart")("Text", {
-  content: Schema_.String
-}) {
-  /**
-   * @since 1.0.0
-   */
-  readonly [PartTypeId]: PartTypeId = PartTypeId
+export type AiInput = typeof AiInput.Type
+
+/**
+ * @since 1.0.0
+ */
+export declare namespace AiInput {
   /**
    * @since 1.0.0
+   * @category Models
    */
-  static fromContent(content: string): TextPart {
-    return new TextPart({ content }, constDisableValidation)
-  }
+  export type Raw =
```

Comment:
In effect 4.0 we should move as many type helper to the top level as possible, to prevent api stutter:

```ts
AiInput.AiInput.Raw
// vs
AiInput.Raw
```